UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common.UnitTests/Helpers/FileShareServiceCacheTest.cs

```diff
@@ -0,0 +1,173 @@
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.Helpers;
+using UKHO.ExchangeSetService.Common.Models.AzureTableEntities;
+using UKHO.ExchangeSetService.Common.Models.FileShareService.Response;
+using UKHO.ExchangeSetService.Common.Models.SalesCatalogue;
+using UKHO.ExchangeSetService.Common.Storage;
+using Attribute = UKHO.ExchangeSetService.Common.Models.FileShareService.Response.Attribute;
+
+namespace UKHO.ExchangeSetService.Common.UnitTests.Helpers
+{
+    public class FileShareServiceCacheTest
+    {
+        private IAzureBlobStorageClient fakeAzureBlobStorageClient;
+        private IAzureTableStorageClient fakeAzureTableStorageClient;
+        private ILogger<FileShareServiceCache> fakeLogger;
+        private ISalesCatalogueStorageService fakeAzureStorageService;
+        private IOptions<CacheConfiguration> fakeCacheConfiguration;
+        private IFileSystemHelper fakeFileSystemHelper;
+        private IFileShareServiceCache fileShareServiceCache;
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeAzureBlobStorageClient = A.Fake<IAzureBlobStorageClient>();
+            fakeAzureTableStorageClient = A.Fake<IAzureTableStorageClient>();
+            fakeLogger = A.Fake<ILogger<FileShareServiceCache>>();
+            fakeAzureStorageService = A.Fake<ISalesCatalogueStorageService>();
+            fakeCacheConfiguration = A.Fake<IOptions<CacheConfiguration>>();
+            fakeFileSystemHelper = A.Fake<IFileSystemHelper>();
+            fakeCacheConfiguration.Value.CacheStorageAccountKey = "testaccountkey";
+            fakeCacheConfiguration.Value.CacheStorageAccountName = "testessstorage";
+            fakeCacheConfiguration.Value.FssSearchCacheTableName = "testfsscache";
+            fakeCacheConfiguration.Value.IsFssCacheEnabled = true;
+
+            fileShareServiceCache = new FileShareServiceCache(fakeAzureBlobStorageClient, fakeAzureTableStorageClient, fakeLogger, fakeAzureStorageService, fakeCacheConfiguration, fakeFileSystemHelper);
+        }
+        private (string, string) GetStorageAccountConnectionStringAndContainerName()
+        {
+            string storageAccountConnectionString = "DefaultEndpointsProtocol = https; AccountName = testessstorage; AccountKey =testaccountkey; EndpointSuffix = core.windows.net";
+            string containerName = "testContainer";
+            return (storageAccountConnectionString, containerName);
+        }
+
+        #region GetProductDetails
+        private List<Products> GetProductdetails()
+        {
+            return new List<Products> {
+                            new Products {
+                                ProductName = "DE416050",
+                                EditionNumber = 2,
+                                UpdateNumbers = new List<int?> {0},
+                                FileSize = 400
+                            }
+                        };
+        }
+        #endregion
```

Comment:
Not a fan of regions so can we lose these directives please:
```suggestion
        private List<Products> GetProductdetails()
        {
            return new List<Products> {
                            new Products {
                                ProductName = "DE416050",
                                EditionNumber = 2,
                                UpdateNumbers = new List<int?> {0},
                                FileSize = 400
                            }
                        };
        }
```